File: frontend/angular/src/app/shared/interfaces/optional-health-profile.ts
export interface OptionalHealthProfile {
    sleepQuality: SleepQuality | null;
    studyLoad: StudyLoad | null;
    smoking: SmokingHabit | null;
    coffee: CoffeeConsumption | null;
    screenTime: ScreenTime | null;
    exercise: ExerciseFrequency | null;
    mealsPerDay: MealsPerDay | null;
    eatSnack: SnackFrequency | null;
    water: WaterIntake | null;
    dietType: DietType | null;
    medication: boolean | null;
    medicationDetails: string | null;
    surgeryHistory: boolean | null;
    surgeryDetails: string | null;
    preferredContent: PreferredContentType | null;
    frequency: ContentFrequency | null;
    comments: string | null;
}

export type OptionalHealthProfileDto = OptionalHealthProfile;

export enum SleepQuality {
    LESS_THAN_6_HOURS = 'LESS_THAN_6_HOURS',
    ABOUT_6_HOURS = 'ABOUT_6_HOURS',
    ABOUT_8_HOURS = 'ABOUT_8_HOURS',
    MORE_THAN_8_HOURS = 'MORE_THAN_8_HOURS'
}

export enum StudyLoad {
    LESS_THAN_1_HOUR = 'LESS_THAN_1_HOUR',
    ONE_TO_TWO_HOURS = 'ONE_TO_TWO_HOURS',
    TWO_TO_FOUR_HOURS = 'TWO_TO_FOUR_HOURS',
    MORE_THAN_4_HOURS = 'MORE_THAN_4_HOURS'
}

export enum SmokingHabit {
    NO = 'NO',
    OCCASIONALLY = 'OCCASIONALLY',
    DAILY = 'DAILY'
}

export enum CoffeeConsumption {
    NONE = 'NONE',
    ONE_CUP = 'ONE_CUP',
    TWO_TO_THREE_CUPS = 'TWO_TO_THREE_CUPS',
    FOUR_OR_MORE_CUPS = 'FOUR_OR_MORE_CUPS'
}

export enum ScreenTime {
    LESS_THAN_2_HOURS = 'LESS_THAN_2_HOURS',
    TWO_TO_FOUR_HOURS = 'TWO_TO_FOUR_HOURS',
    FOUR_TO_SIX_HOURS = 'FOUR_TO_SIX_HOURS',
    MORE_THAN_6_HOURS = 'MORE_THAN_6_HOURS'
}

export enum ExerciseFrequency {
    NONE = 'NONE',
    ONE_TO_TWO_TIMES = 'ONE_TO_TWO_TIMES',
    THREE_TO_FIVE_TIMES = 'THREE_TO_FIVE_TIMES',
    FIVE_TO_SEVEN_TIMES = 'FIVE_TO_SEVEN_TIMES'
}

export enum MealsPerDay {
    ONE_TO_TWO = 'ONE_TO_TWO',
    THREE = 'THREE',
    FOUR_TO_FIVE = 'FOUR_TO_FIVE',
    MORE_THAN_FIVE = 'MORE_THAN_FIVE'
}

export enum SnackFrequency {
    NO = 'NO',
    SOMETIMES = 'SOMETIMES',
    DAILY = 'DAILY'
}

export enum WaterIntake {
    ONE_TO_THREE_GLASSES = 'ONE_TO_THREE_GLASSES',
    FOUR_TO_SIX_GLASSES = 'FOUR_TO_SIX_GLASSES',
    SEVEN_TO_NINE_GLASSES = 'SEVEN_TO_NINE_GLASSES',
    MORE_THAN_NINE_GLASSES = 'MORE_THAN_NINE_GLASSES'
}

export enum DietType {
    OMNIVORE = 'OMNIVORE',
    VEGETARIAN = 'VEGETARIAN',
    VEGAN = 'VEGAN',
    PESCATARIAN = 'PESCATARIAN',
    GLUTEN_FREE = 'GLUTEN_FREE',
    OTHER = 'OTHER'
}

export enum PreferredContentType {
    ARTICLES = 'ARTICLES',
    SHORT_TIPS = 'SHORT_TIPS',
    MIXED = 'MIXED'
}

export enum ContentFrequency {
    DAILY = 'DAILY',
    FEW_TIMES_PER_WEEK = 'FEW_TIMES_PER_WEEK',
    RARELY = 'RARELY'
}

export enum YesNoOption {
    NO = 'NO',
    YES = 'YES'
}

File: backend/src/main/java/gr/uniwa/unihealth/backend/model/OptionalHealthProfile.java
package gr.uniwa.unihealth.backend.model;

import gr.uniwa.unihealth.backend.model.enums.optionalFormFieldEnum.*;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "t_optional_health_profile")
public class OptionalHealthProfile extends BaseUpdatableEntity {

  @OneToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "health_profile_id", nullable = false, unique = true)
  private HealthProfile healthProfile;

  @Enumerated(EnumType.STRING)
  @Column(name = "sleep_quality")
  private SleepQuality sleepQuality;

  @Enumerated(EnumType.STRING)
  @Column(name = "study_load")
  private StudyLoad studyLoad;

  @Enumerated(EnumType.STRING)
  private SmokingHabit smoking;

  @Enumerated(EnumType.STRING)
  private CoffeeConsumption coffee;

  @Enumerated(EnumType.STRING)
  @Column(name = "screen_time")
  private ScreenTime screenTime;

  @Enumerated(EnumType.STRING)
  private ExerciseFrequency exercise;

  @Enumerated(EnumType.STRING)
  @Column(name = "meals_per_day")
  private MealsPerDay mealsPerDay;

  @Enumerated(EnumType.STRING)
  @Column(name = "eat_snack")
  private SnackFrequency eatSnack;

  @Enumerated(EnumType.STRING)
  private WaterIntake water;

  @Enumerated(EnumType.STRING)
  @Column(name = "diet_type")
  private DietType dietType;

  private Boolean medication;

  @Column(name = "medication_details")
  private String medicationDetails;

  @Column(name = "surgery_history")
  private Boolean surgeryHistory;

  @Column(name = "surgery_details")
  private String surgeryDetails;

  @Enumerated(EnumType.STRING)
  @Column(name = "preferred_content")
  private PreferredContentType preferredContent;

  @Enumerated(EnumType.STRING)
  private ContentFrequency frequency;

  private String comments;
}
